native/cocos/platform/filesystem/IFileHandle.h

```diff
@@ -0,0 +1,53 @@
+
+/****************************************************************************
+ Copyright (c) 2022 Xiamen Yaji Software Co., Ltd.
+ 
+ http://www.cocos.com
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+ 
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+
+#pragma once
+
+#include "base/Macros.h"
+#include "base/TypeDef.h"
+
+namespace cc {
+
+class CC_DLL IFileHandle {
+public:
+    enum class MoveMethod{
+        FILE_SEEK_CUR = 0,
+        FILE_SEEK_END = 1,
+        FILE_SEEK_SET = 2,
+    };
+    IFileHandle() = default;
```

Comment:
I think the constructor can be private, as it seems meaningless to new a IFileHandle in codes.

Reply:
This is a pure virtual function and cannot be constructed.

Reply:
Yep, we should forbid its children too.

Reply:
Sorry, i mean its sub class.

Reply:
Yes, derived classes are appropriate, but friend classes may be used, created in FileSystem